### PR TITLE
Initial push

### PR DESCRIPTION
**Pull request by Cappo**
Contains the code for the `vela-npm` plugin for publishing Node.JS packages.

Basic usage:
```
steps:
  - name: publish to npm
    image: target/vela-npm
    secrets: [ npm_token ]
```

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at
+[TTS-OpenSource-Office@target.com](mailto:TTS-OpenSource-Office@target.com). All
```
Comment (kaymckay) [inline .github/CODE_OF_CONDUCT.md:59]: I believe the updated email is opensource@target.com

File: build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2022 Target Brands, Inc. All rights reserved.
+#
+# Use of this source code is governed by the LICENSE file in this repository.
+
+###########################################################################
+##    docker build --no-cache -t vela-npm:local -f build/Dockerfile .    ##
+###########################################################################
+
```
Comment (kaymckay) [inline build/Dockerfile:8]: Not a big deal, but it looks like every other repo keeps the Dockerfile at the top level

File: cmd/vela-npm/main.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	"os"
+	"strings"
+
+	"github.com/go-vela/vela-npm/internal/npm"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+func main() {
+	app := &cli.App{
```
Comment (kaymckay) [inline cmd/vela-npm/main.go:16]: There is some standard plugin information to include before app creation
```
        // capture application version information.
	pluginVersion := version.New()
	// serialize the version information as pretty JSON
	bytes, err := json.MarshalIndent(pluginVersion, "", "  ")
	if err != nil {
		logrus.Fatal(err)
	}

	// output the version information to stdout
	fmt.Fprintf(os.Stdout, "%s\n", string(bytes))
```

File: cmd/vela-npm/main.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+		Name:      "vela-npm",
+		Usage:     "Vela npm plugin for publishing NodeJS packages",
+		Version:   "2.1.0",
```
Comment (kaymckay) [inline cmd/vela-npm/main.go:19]: version becomes:
``
Version:  pluginVersion.Semantic(),
``

**Review by kaymckay — CHANGES_REQUESTED** [review]
Commented on some little things, but other than that looks good 👍  I noticed the workflows didn't run because you are PRing from a forked repo, so I would run golangci, that one always complains about something

File: build/Dockerfile (hunk 2 above)
Comment (Cappo) [reply build/Dockerfile]: Moved it tot he root to keep it consistent

File: cmd/vela-npm/main.go (hunk 4 above)
Comment (Cappo) [reply cmd/vela-npm/main.go]: I don't have any experience with the version library the other plugins are referencing, but used Kaniko as an example to update this one.

**Review by kaymckay — APPROVED** [review]
Looks good, thanks for the changes 👍

**Comment by kaymckay** [comment]
@Cappo Can you add the changelog folder w/ config file like https://github.com/go-vela/vela-kaniko/tree/master/.chglog

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+require (
+	github.com/Masterminds/semver/v3 v3.1.1
+	github.com/cpuguy83/go-md2man/v2 v2.0.1 // indirect
+	github.com/go-vela/types v0.11.0
+	github.com/golang/mock v1.6.0
+	github.com/sirupsen/logrus v1.8.1
+	github.com/spf13/afero v1.8.0
+	github.com/urfave/cli/v2 v2.3.0
+	golang.org/x/sys v0.0.0-20220111092808-5a964db01320 // indirect
+	golang.org/x/text v0.3.7 // indirect
```
Comment (wass3r) [inline go.mod:14]: there are a few dependencies out of date here including go itself. slightly leaning towards taking care of that now vs after this initial merge.

Comment (GregoryDosh) [reply go.mod]: @Cappo Is taking some well earned vacation time ATM so I bumped these deps for him.

**Review by kaymckay — DISMISSED** [review]
👍

**Review by kaymckay — APPROVED** [review]

**Review by wass3r — APPROVED** [review]
thanks

**Pull request merged**